src/test/java/org/folio/linked/data/e2e/resource/ResourceControllerRetainOutgoingEdgesIT.java

```diff
@@ -24,34 +27,19 @@
 import org.folio.linked.data.domain.dto.ResourceResponseDto;
 import org.folio.linked.data.e2e.base.IntegrationTest;
 import org.folio.linked.data.model.entity.FolioMetadata;
-import org.folio.linked.data.model.entity.Resource;
-import org.folio.linked.data.model.entity.ResourceEdge;
-import org.folio.linked.data.service.resource.hash.HashService;
-import org.folio.linked.data.test.ResourceTestService;
+import org.folio.linked.data.test.kafka.KafkaProducerTestConfiguration;
 import org.junit.jupiter.api.Test;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.core.env.Environment;
-import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
 
 @IntegrationTest
-class ResourceControllerRetainOutgoingEdgesIT {
-  private static final String RESOURCE_URL = "/linked-data/resource";
-  private static final String WORK_ID_PLACEHOLDER = "%WORK_ID%";
-
-  @Autowired
-  private ResourceTestService resourceTestService;
-  @Autowired
-  private Environment env;
-  @Autowired
-  private MockMvc mockMvc;
-  @Autowired
-  private HashService hashService;
+@SpringBootTest(classes = {KafkaProducerTestConfiguration.class})
```

Comment:
This test was not working properly without @SpringBootTest annotation, however it was green after 1 min timeout:
<img width="450" alt="image" src="https://github.com/user-attachments/assets/314f99b1-7e0b-429d-b5f4-385092d75320" />
<img width="450" alt="image" src="https://github.com/user-attachments/assets/53fa92cd-cc67-4335-a0c3-232f0124e413" />